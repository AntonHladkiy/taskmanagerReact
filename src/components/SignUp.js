import React, { useState } from 'react';
import {Link} from "react-router-dom";
const SignUp = props => {
    const [user, setUser] = useState(props.initialUser);

    const handleInputChange = event => {
        const { name, value } = event.target
        setUser({ ...user, [name]: value })
    };

    return (
        <form className={"form-check"} autoComplete="off">
            <label className= "form-check-label">Email</label>
            <input className= "form-control w-50" type="text" name="email" value={user.email} onChange={handleInputChange} ></input>
            <label className= "form-check-label">First Name</label>
            <input className= "form-control w-50" type="text" name="firstName" value={user.firstName} onChange={handleInputChange} ></input>
            <label className= "form-check-label">Last Name</label>
            <input className= "form-control w-50" type="text" name="lastName" value={user.lastName} onChange={handleInputChange} ></input>
            <label className= "form-check-label">Password</label>
            <input className= "form-control w-50" name="password" type="password" value={user.password} onChange={handleInputChange} ></input>
            <label className= "form-check-label">Password</label>
            <input className= "form-control w-50" type="password" name="passwordConfirmation" value={user.passwordConfirmation} onChange={handleInputChange} ></input>
            <Link to ="/login"> <button className="btn btn-success mt-2 mr-2 w-25" onClick={()=> {
                if (!user.email|| !user.password|| !user.passwordConfirmation|| !user.lastName|| !user.firstName|| (user.passwordConfirmation!==user.password)) {
                    alert("Wrong form format try again")
                    return
                };
                props.signUp(user)
            }}>Sign Up</button></Link>
            <Link to ="/"><button  className={"btn btn-danger mt-2 ml-auto w-25"}>Cancel</button></Link>
        </form>
    )
}
export default SignUp;