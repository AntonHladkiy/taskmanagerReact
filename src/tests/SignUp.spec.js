import React from 'react';
import { shallow} from 'enzyme';
import SignUp from '../components/SignUp';
let initialUser={
    email:'',
    password:''
}
describe('SignUp', () => {
    let wrapped = shallow(<SignUp initialUser={initialUser} />);
    it('should render the SignUp Component correctly', () => {
        expect(wrapped).toMatchSnapshot();
    });
});