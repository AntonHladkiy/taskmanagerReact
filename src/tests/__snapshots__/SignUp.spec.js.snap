// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp should render the SignUp Component correctly 1`] = `
<form
  autoComplete="off"
  className="form-check"
>
  <label
    className="form-check-label"
  >
    Email
  </label>
  <input
    className="form-control w-50"
    name="email"
    onChange={[Function]}
    type="text"
    value=""
  />
  <label
    className="form-check-label"
  >
    First Name
  </label>
  <input
    className="form-control w-50"
    name="firstName"
    onChange={[Function]}
    type="text"
  />
  <label
    className="form-check-label"
  >
    Last Name
  </label>
  <input
    className="form-control w-50"
    name="lastName"
    onChange={[Function]}
    type="text"
  />
  <label
    className="form-check-label"
  >
    Password
  </label>
  <input
    className="form-control w-50"
    name="password"
    onChange={[Function]}
    type="password"
    value=""
  />
  <label
    className="form-check-label"
  >
    Password confirmation
  </label>
  <input
    className="form-control w-50"
    name="passwordConfirmation"
    onChange={[Function]}
    type="password"
  />
  <Link
    to="/login"
  >
     
    <button
      className="btn btn-success mt-2 mr-2 w-25"
      onClick={[Function]}
    >
      Sign Up
    </button>
  </Link>
  <Link
    to="/"
  >
    <button
      className="btn btn-danger mt-2 ml-auto w-25"
    >
      Cancel
    </button>
  </Link>
</form>
`;
